### DISCUSSION TITLE
Error while using polygon zone

### DISCUSSION THREAD

**Discussion by uzumakinaruto19** [roboflow/supervision#133] [Q&A] [answered]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar bug report.

### Bug

When I was trying to use the polygon zone and trying to trigger the count, the output video runs for sometime and throws this error

`    result[:, [0, 2]] = result[:, [0, 2]].clip(0, width)
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
`

### Environment

- Supervision: 0.9.0
- OS : Ubuntu 22
- Python: 3.9

### Minimal Reproducible Example

```python

`import cv2

from ultralytics import YOLO
import supervision as sv
import numpy as np

polygon = np.array([
    [409, 226],
    [327, 235],
    [327, 204],
    [398, 198],
])
video_info = sv.VideoInfo.from_video_path('./test.mp4')
zone = sv.PolygonZone(polygon=polygon, frame_resolution_wh=video_info.resolution_wh)
`
def main():
    global model

    zone_annotator = sv.PolygonZoneAnnotator(zone=zone, color=sv.Color.white(), thickness=1, text_thickness=1, text_scale=0.5)

    box_annotator = sv.BoxAnnotator(
        thickness=2,
        text_thickness=1,
        text_scale=0.5
    )
  
    model = YOLO("yolov8n.pt")

    for result in model.track(source='./test.mp4', classes=0, imgsz= 480, show=True, stream=True,
                            agnostic_nms=True):
        
        frame = result.orig_img
        detections = sv.Detections.from_yolov8(result)

        if result.boxes.id is not None:
            detections.tracker_id = result.boxes.id.cpu().numpy().astype(int)

        else:
            detections.tracker_id = np.array([])
            detections.conf = np.array([])
            detections.xyxy = np.array([])
        
        # detections = detections[(detections.class_id != 60) & (detections.class_id != 0)]

        labels=[str(id) for id in detections.tracker_id]

        frame = box_annotator.annotate(
            scene=frame, 
            detections=detections,
            labels=labels
        )
     
        zone.trigger(detections=detections)

        zone_annotator.annotate(scene=frame)
       
        print(output)

        cv2.imshow("yolov8", frame)

        if (cv2.waitKey(30) == 27):
            break


if __name__ == "__main__":
    main()
```

### Additional

same code works fine with Linezone, I'm trying to do both in my code

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by SkalskiP** [comment]
Hi @uzumakinaruto19 👋🏻 ! Could you give me a full stack trace? I'm pretty sure it is not full error message:

```
result[:, [0, 2]] = result[:, [0, 2]].clip(0, width) IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
```

**Comment by SkalskiP** [comment]
Also, can you share `test.mp4` ?

**Comment by uzumakinaruto19** [comment]
> Hi @uzumakinaruto19 👋🏻 ! Could you give me a full stack trace? I'm pretty sure it is not full error message:
> 
> ```
> result[:, [0, 2]] = result[:, [0, 2]].clip(0, width) IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed
> 
```

video 1/1 (117/2699) ~/Revamped/webrtc-streaming/test.mp4: 288x480 1 person, 35.3ms
{'frame_index': 0, 'time_stamp': 0, 'detections': [[0, 164, 79, 406]], 'demo_run': False, 'ids': [1]}
video 1/1 (118/2699) ~/Desktop/Revamped/webrtc-streaming/test.mp4: 288x480 1 person, 32.4ms
Traceback (most recent call last):
  File "~/DesktopRevamped/webrtc-streaming/test.py", line 93, in <module>
    main()
  File "~/Desktop/Revamped/webrtc-streaming/test.py", line 64, in main
    zone.trigger(detections=detections)
  File "~/anaconda3/envs/lobby/lib/python3.9/site-packages/supervision/detection/tools/polygon_zone.py", line 54, in trigger
    clipped_xyxy = clip_boxes(
  File "~/anaconda3/envs/lobby/lib/python3.9/site-packages/supervision/detection/utils.py", line 117, in clip_boxes
    result[:, [0, 2]] = result[:, [0, 2]].clip(0, width)
IndexError: too many indices for array: array is 1-dimensional, but 2 were indexed

**Comment by uzumakinaruto19** [comment]
> Also, can you share `test.mp4` ?

I cant share the video, but I have tried with 2 videos both failed around 118 and 230th frame

**Comment by SkalskiP** [comment]
It will be hard for me to help you if I can't reproduce the error.

Can you change your code like this 👇🏻 ? We want to wrap `zone.trigger` into this try/catch statement and see if the value of `detections.xyxy` is okay.

```python
try:
    zone.trigger(detections=detections)
catch IndexError:
    print(detections.xyxy)
    print(detections.xyxy.shape)
```

**Comment by uzumakinaruto19** [comment]
it worked when I put a try and except statement
Thanks alot

**Comment by SkalskiP** [comment]
This is not a solution :)) This is just a workaround. I assume there was some output/log when you executed it. Could you paste it here?

**Comment by uzumakinaruto19** [comment]
> This is not a solution :)) This is just a workaround. I assume there was some output/log when you executed it. Could you paste it here?

I didn't try catch Indexerror statement,

`[] error in xyxy
(0,) error in shape`

these are the output from the prints statements

I think it happened while the detection is empty

**Comment by SkalskiP** [comment]
Yup, it looks like it is the case. I'll work on the fix today. But it won't be accessible in `supervision` until the `0.10.0` release.

**Comment by SkalskiP** [comment]
Okay, I know what's up. It looks like the problem is in your code. Please change:

```python
if result.boxes.id is not None:
    detections.tracker_id = result.boxes.id.cpu().numpy().astype(int)

else:
    detections.tracker_id = np.array([])
    detections.conf = np.array([])
    detections.xyxy = np.array([])
```

Into:

```python
if result.boxes.id is not None:
    detections.tracker_id = result.boxes.id.cpu().numpy().astype(int)

else:
    detections.tracker_id = np.array([])
```

I don't think you need.

```python
detections.conf = np.array([])
detections.xyxy = np.array([])
```

**Comment by uzumakinaruto19** [comment]
hey, I tried that, and got a new error,

`    self.tracker_id[i] if self.tracker_id is not None else None,
IndexError: index 0 is out of bounds for axis 0 with size 0
`
now throw error in line_counter.trigger() function

`
 File "~/webrtc-streaming/detection_test.py", line 70, in main

    line_counter_1.trigger(detections=detections)  
 
 File "~/webrtc-streaming/counter.py", line 44, in trigger

    for xyxy, _, confidence, class_id, tracker_id in detections:

File "~/anaconda3/envs/lobby/lib/python3.9/site-packages/supervision/detection/core.py", line 108, in __iter__

    self.tracker_id[i] if self.tracker_id is not None else None,

IndexError: index 0 is out of bounds for axis 0 with size 0
`

**Comment by SkalskiP** [comment]
Since, at this stage, we are discussing details related to your code and not potential problems with the library, I am converting this issue into a discussion.

**Comment by SkalskiP** [comment]
@uzumakinaruto19 okey so in that case do it like this:

```python
if result.boxes.id is not None:
    detections.tracker_id = result.boxes.id.cpu().numpy().astype(int)

else:
    detections.tracker_id = np.array([])
    detections.conf = np.array([])
    detections.xyxy=np.empty((0, 4), dtype=np.float32)
```

**Reply by Archangel31** [reply]
Hey, I am getting a similar issue, Can you please look into this:
**Code**:

for frame_num, detection in enumerate(detections):

            cls_names = detection.names
            cls_names_inv = {v:k for k,v in cls_names.items()}
            
            detection_supervision = sv.Detections.from_ultralytics(detection)
            
            for object_ind , class_id in enumerate(detection_supervision.class_id):
                if cls_names[class_id] == "goalkeeper":
                    detection_supervision.class_id[object_ind] = cls_names_inv["player"]
            
           
            detection_with_tracks = self.tracker.update_with_detections(detection_supervision)
            
            tracks["players"].append({})
            tracks["referees"].append({})
            tracks["ball"].append({})
            
            for frame_detection in detection_with_tracks:
                bbox = frame_detection[0].tolist()
                cls_id = frame_detection[3]
                track_id = frame_detection[4]
                
                
**Error:**                
    self.tracker_id[i] if self.tracker_id is not None else None,
IndexError: index 0 is out of bounds for axis 0 with size 0

**Reply by LinasKo** [reply, marked as answer]
Hi @Archangel31 

Can you please try this supervision version?

```bash
pip install supervision==0.21.0.rc5
```

**Reply by Archangel31** [reply]
Thank you so much! It worked.
